submissions/ses3332002/memory-game/script.js

```diff
@@ -0,0 +1,140 @@
+"use strict";
+  
+  const doc = document.body;
+  const themeButtons = document.querySelector(".theme_selector");
+  const sizeSelector = document.querySelector(".size_selector");
+  const cardThemeSelector = document.querySelectorAll('input[name="option"]');
+  const startButton = document.querySelector(".start_button");
+  const gameArea = document.querySelector(".game_area");
+  const switchableControls = document.querySelectorAll(".switchable");
+  let numbersArray;
+  let cardThemeSelected;
+  let cardSelected;
+  let cardsArray;
+  let matchedCounter;
+  
+  startButton.addEventListener("click", startGame);
+  doc.classList.add("theme1");
+  
+  themeButtons.addEventListener("click", function({target}) {
+    if (target.tagName != "BUTTON") {
+      return;
+    };
+    doc.classList.remove(doc.classList[0]);
+    doc.classList.add(target.dataset.theme);
+  });
+  
+  function startGame() {
+    cardThemeSelected = document.querySelector('input[name="option"]:checked');
+    matchedCounter = 0;
+    prepareGameField();
+    switchControls();
+  };
+  
+  function switchControls() {
+    switchableControls.forEach((item) => {
+      item.disabled = !item.disabled;
+      item.classList.toggle("form_item-desabled");
+    });
+  };
+  
+  function prepareGameField() {
+    numbersArray = [];
+    for (let i=0; i < sizeSelector.value/2; i++) {
+      numbersArray[i] = i;
+    };
+    numbersArray = numbersArray.concat(numbersArray);
+
+    function shuffle(array) {
+      for (let i = array.length - 1; i > 0; i--) {
+        let j = Math.floor(Math.random() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
+      };
+    };
+    
+    shuffle(numbersArray);
+    
+    if (document.documentElement.clientHeight > document.documentElement.clientWidth) {
```

Comment:
I'd recommend to move this big check to the separate function. It would be easier to read your code

Reply:
done